### DISCUSSION TITLE
Write to code permissions

### DISCUSSION THREAD

**Discussion by laurentsimon** [renovatebot/renovate#15814] [Request Help]
### What would you like Renovate to be able to do?

Renovatebot askes for `write` permission to code `Read and write access to checks, code, commit statuses, issues, pull requests, and workflows`.

I suppose that's for the auto-merge feature?

I would like to have a renovatebot app that only asks for `read` permissions. I wanted to move away from dependabot in favor of renovatebot, but the write permission is a no-starter.

### If you have any ideas on how this should be implemented, please tell us here.

GH app don't allow dynamic permissions, so I suppose the only way to achieve this is to have2 different apps. There are probably usability concerns around this though.

### Is this a feature you are interested in implementing yourself?

No

**Comment by rarkins** [comment]
An app needs write permissions just to create branches actually. But check out the Forking Renovate app - it needs only read-only code access and submits PRs from a fork. It's used by most Google-run orgs

**Comment by laurentsimon** [comment]
I'll check the Forking Renovate app, thank you!

**Reply by HonkingGoose** [reply]
For more information see the [Renovate docs, Security and Permissions](https://docs.renovatebot.com/security-and-permissions/).

**Comment by viceice** [comment]
converted to discussion, as it's nothing the oss cli can implement 😉
